scanner/src/main/kotlin/experimental/ExperimentalScanner.kt

```diff
@@ -421,6 +428,51 @@ class ExperimentalScanner(
             ScanResult(provenance, scanner.details, summary)
         }
     }
+
+    private fun createFileArchives(nestedProvenances: Map<Package, NestedProvenance>) {
+        // TODO: The archives are currently created in a way compatible with the existing implementation in the
+        //       LocalScanner. This allows to keep using existing file archives without changing the logic used to
+        //       access those archives in the reporter. To achieve this nested provenances are downloaded recursively,
+        //       so that the created archives contain also files from nested repositories.
+        //       This could be replaced with creating file archives for each provenance separately and building the
+        //       final result on demand, to reduce duplication in the file archives.
+
+        val archiver = scannerConfig.archive.createFileArchiver()
+
+        val provenancesWithMissingArchives = nestedProvenances.filterNot { (_, nestedProvenance) ->
+            archiver.hasArchive(nestedProvenance.root)
+        }
+
+        log.info { "Creating file archives for ${provenancesWithMissingArchives.size} packages." }
+
+        val duration = measureTime {
+            provenancesWithMissingArchives.forEach { (_, nestedProvenance) ->
+                val dir = downloadRecursively(nestedProvenance)
+                archiver.archive(dir, nestedProvenance.root)
+                dir.safeDeleteRecursively(force = true)
+            }
+        }
+
+        log.info {
+            "Created file archives for ${provenancesWithMissingArchives.size} packages in " +
+                    "${duration.inWholeMilliseconds} ms."
```

Comment:
Please don't use an explicit unit anymore, but let the `Duration` class itself do the formatting. Also see https://github.com/oss-review-toolkit/ort/pull/4573.

Reply:
Thanks, I didn't notice #4573. It's nice for the end-user but it makes log analytics for performance very difficult. I have changed it for consistency but I think we should reconsider this sometime.